src/features/snap/components/MetadataAuditItem.tsx

```diff
@@ -0,0 +1,49 @@
+import { Link } from '@chakra-ui/react';
+import { t } from '@lingui/macro';
+import { navigate } from 'gatsby';
+import type { FunctionComponent } from 'react';
+import type { Hex } from 'viem';
+import { mainnet, useEnsName } from 'wagmi';
+
+import { Data } from './Data';
+import { trimAddress } from '../../../utils';
+
+export type MetadataAuditItemProps = {
+  auditorAddresses: Hex[];
+};
+
+export const MetadataAuditItem: FunctionComponent<MetadataAuditItemProps> = ({
+  auditorAddresses,
+}) => {
+  const auditors: any[] = [];
+  for (const address of auditorAddresses) {
+    // eslint-disable-next-line react-hooks/rules-of-hooks
```

Comment:
We must avoid this at all cost.  
We should use a dedicated component to display an address (with or without ENS).  
I've created this component (not merged yet): https://github.com/MetaMask/permissionless-snaps-directory/pull/157/files#diff-5ce11a0e92da904e67ddb816abcdbf5553a5972d7fd4745df0de2257e95961e3
This might be helpful, wdyt?